tests/_util/test_template_util.py

```diff
@@ -0,0 +1,141 @@
+                Parameter(
+                    name = "param 1",
+                    value = NormalReal(mean=4.2, std = 0.1, units="g")
+                ),
+                Parameter(
+                    name = "param 3",
+                    value = NominalCategorical(category="bar")
+                )
+            ],
+            conditions = [
+                Condition(
+                    name = "cond 1",
+                    value = NormalReal(mean=4, std=0.5, units="")
+                ),
+                Condition(
+                    name = "cond 2",
+                    value = NominalCategorical(category="hi")
+                ),
+                Condition(
+                    name = "attr 1",
+                    value = InChI(inchi="InChI=1S/C34H34N4O4.Fe/c1-7-21-17(3)25-13-26-19(5)23(9-11-33(39)40)31(37-26)16-32-24(10-12-34(41)42)20(6)28(38-32)15-30-22(8-2)18(4)27(36-30)14-29(21)35-25;/h7-8,13-16H,1-2,9-12H2,3-6H3,(H4,35,36,37,38,39,40,41,42);/q;+2/p-2")
+                ),
+            ]
+        ),
+        MaterialSpec(
+            name = "material 1",
+            process = LinkByUID(scope = "faux 2", id = "id2"),
+            properties=[
+                PropertyAndConditions(
+                    property=Property(
+                        name = "prop 1",
+                        value = NormalReal(mean=100, std=10, units="g/cm**3")
+                    ),
+                    conditions=[
+                        Condition(
+                        name = "cond 2",
+                        value = NominalCategorical(category="hi")
+                        )
+                    ]
+                ),
+                PropertyAndConditions(
+                    property=Property(
+                        name = "prop 2",
+                        value = NominalReal(nominal=33, units="1/lb")
+                    ),
+                    conditions=[
+                        Condition(
+                        name = "cond 3",
+                        value = NominalCategorical(category="citrine")
+                        )
+                    ]
+                ),
+            ]
+        ),
+        MeasurementSpec(
+            name = "meas spec 1",
+            parameters = [
+                Parameter(
+                    name = "param 1",
+                    value = NominalReal(nominal=2.2, units="kg")
+                ),
+                Parameter(
+                    name = "param 2",
+                    value = NominalCategorical(category="bar")
+                )
+            ],
+        ),
+        MeasurementRun(
+            name = "meas run 1",
+            spec = LinkByUID(scope="another fake scope", id = "another fake id"),
+            properties=[
+                Property(
+                    name = "prop 1",
+                    value=NominalReal(nominal=4.1, units="")
+                )
+            ]
+        )
+    ]
+    return faux_gems
+
+def test_attribute_alignment():
```

Comment:
I think it would be good to have a test that lights up the `flatten`.  The simplest unit would be to just have two linked objects (e.g. process run and process spec) and only pass the process run in.

Reply:
I do have this in the tests. Line 41 is where I have my nestled Object. The shape assertion confirms it has been flattened. Do you feel this is inadequate or should be more standalone?

Reply:
No, that covers it.  I just didn't see it.